### DISCUSSION TITLE
Helix online playground

### DISCUSSION THREAD

**Discussion by tomgroenwoldt** [helix-editor/helix#7205] [Show and tell]
I experimented with [xtermjs](xtermjs.org) and created a simple terminal frontend which is attached to a websocket serving a real `helix` instance.

![helix-editor-playground](https://s11.gifyu.com/images/SQXPy.gif)

The websocket also resizes the real terminal running `helix` whenever the client resizes his web terminal. Check it out [here](https://github.com/tomgroenwoldt/helix-editor-playground) and tell me about your thoughts :)

**Comment by href** [comment]
This is fun! What kind of host am I connecting to with this? It looks like this is running in a container of sorts?

In any case, maybe you should disable network access, I was able to use `curl`. With that I'm only a few commands away from running my own binaries, which is probably not what you want.

**Reply by tomgroenwoldt** [reply]
The backend runs on a small server I rent. Yes, you are right, every user gets his own small container for this. Thanks for catching the network access! :) This was not intended.

**Reply by tomgroenwoldt** [reply]
Thanks to your suggestion I figured that removing the users access to `bash` should solve the overall problem. Even without network access one could e.g. `base64` encode binaries and paste them directly as a ready file via `:sh base64 --decode ...`. If there is no bash, the `helix` `:sh` command is unusable.

**Reply by rozaliev** [reply]
@tomgroenwoldt Removing bash is not enough, you should probably have as minimal of a container as possible and as low privileges as possible, don't rely on Helix for sandboxing.
![image](https://github.com/helix-editor/helix/assets/120809/2a983acd-18ff-4d60-b476-0f7e24bef71a)

**Reply by tomgroenwoldt** [reply]
@rozaliev That's a good point. I reworked the setup of the application and open-sourced everything in the above mentioned repository :)

**Comment by rkshthrmsh** [comment]
@tomgroenwoldt, this is quite cool! Was wondering what version of Helix is running on the server? Just a thought: Having the latest master version running would allow users to test out yet-to-be-released features on this playground. One example is of the code-tree which is still pending review.

**Reply by tomgroenwoldt** [reply]
The version should be `23.05-1`. This is a cool idea, maybe I'm gonna introduce the ability to switch between the stable release and master. Also if you have anything specific in mind feel free to open an issue :)

**Reply by tomgroenwoldt** [reply]
@rkshthrmsh this is now possible and updated daily: https://tomgroenwoldt.github.io/helix-editor-playground/
